dali/operators/generic/expand_dims.cc

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include <vector>
+#include <set>
+
+#include "dali/core/math_util.h"
+#include "dali/core/static_switch.h"
+#include "dali/core/tensor_layout.h"
+#include "dali/core/tensor_shape_print.h"
+#include "dali/operators/generic/expand_dims.h"
+#include "dali/pipeline/data/views.h"
+
+namespace dali {
+
+DALI_SCHEMA(ExpandDims)
+  .DocStr(R"code(Insert new dimension[s] of extent 1 and inserts new entries in "
+    "the layout (new_axis_names) at these indices in the layout.)code")
+  .NumInput(1)
+  .NumOutput(1)
+  .InputDox(0, "data", "TensorList", "Data to be expanded")
+  .PassThrough({{0, 0}})
+  .AllowSequences()
+  .SupportVolumetric()
+  .AddOptionalArg<int>("axes", R"code(Indices where to put new dimensions of size 1.)code",
+    std::vector<int>(), true)
+  .AddOptionalArg("new_axis_names", R"code(Names of new dimensions in data layout.
+  
+Size of ``new_axis_names`` should be equal to ``axe``s size. 
+If argument won't be provided new dimensions will have layout '?')code", TensorLayout(""));
```

Comment:
I wonder if it wouldn't be better to clear the layout altogether... Let's discuss it.